FixMatrix/FixMatrix.cpp

```diff
@@ -0,0 +1,208 @@
+#include <cstdlib>
+#include <iostream>
+#include <stdexcept>
+
+namespace {
+
+    class matrix {
+        int Row, Col;
+        double **Value;
+    public:
+
+        matrix(unsigned int, unsigned int);
+
+        matrix(const matrix &);
+
+        int GetRow();
+
+        int GetCol();
+
+
+        friend bool operator==(const matrix &m1, const matrix &m2);
+
+        friend bool operator!=(const matrix &m1, const matrix &m2);
+
+        friend std::istream &operator>>(std::istream &istr, matrix &m);
+
+        friend std::ostream &operator<<(std::ostream &ostr, matrix &m);
+
+        friend matrix operator+(const matrix &m1, const matrix &m2);
+
+        friend matrix operator-(const matrix &m1, const matrix &m2);
+
+        friend matrix operator*(const matrix &m1, const matrix &m2);
+    };
+
+
+    bool operator==(const matrix &m1, const matrix &m2) {
+        return m1.Row == m2.Row && m1.Col == m2.Col;
+    }
+
+    bool operator!=(const matrix &m1, const matrix &m2) {
+        return !(m1 == m2);
+
+    }
+
+
+    ~
+
+    matrix();
+};
+
+matrix::matrix(int row, int col) {
+    Row = row;
+    Col = col;
+    Value = new double *[row];
+    for (int i = 0; i < row; i++) Value[i] = new double[col];
+}
+
+matrix::matrix(const matrix &m)
+        : Row(m.Row),
+          Col(m.Col) {
+    Value = new double *[Row];
+    for (int i = 0; i < Row; i++) Value[i] = new double[Col];
+    for (int i = 0; i < Row; i++) {
+        for (int j = 0; j < Col; j++)
+            Value[i][j] = m.Value[i][j];
+    }
+}
+
+int matrix::GetRow() {
+    return Row;
+}
+
+int matrix::GetCol() {
+    return Col;
+}
+
+std::istream &operator>>(std::istream &istr, matrix &m) {
+    for (int i = 0; i < m.GetRow(); i++)
+        for (int j = 0; j < m.GetCol(); j++)
+            istr >> m(i, j);
+    return (istr);
+}
+
+std::ostream &operator<<(std::ostream &ostr, matrix &m) {
+    for (int i = 0; i < m.GetRow(); i++) {
+        for (int j = 0; j < m.GetCol(); j++)
+            ostr << m(i, j) << "\t";
+        ostr << "\n";
+    }
+    return (ostr);
+}
+
+matrix operator+(matrix & m1, matrix & m2) {
```

Comment:
& to the left side like this:
`const matrix& m`

not like this
`const matrix &m`

and definitely not like this
`const matrix & m`